client/src/webview-js/image-upload/image-upload.ts

```diff
@@ -1,22 +1,30 @@
-let vscode
-let pending = []
+declare let acquireVsCodeApi: any
+let vscode: any
+
+interface PendingEntry {
+  mediaName: string
+  data: string | ArrayBuffer | null
+}
+let pending: PendingEntry[] = []
 
 window.addEventListener('load', () => {
   vscode = acquireVsCodeApi() // eslint-disable-line no-undef
 
-  const dropArea = document.getElementById('drop-area')
-  const preview = document.getElementById('preview')
-  const uploadButton = document.getElementById('trigger-upload')
-  const cancelButton = document.getElementById('cancel-upload')
+  const dropArea = document.getElementById('drop-area') as HTMLElement
+  const preview = document.getElementById('preview') as HTMLElement
+  const uploadButton = document.getElementById('trigger-upload') as HTMLElement
+  const cancelButton = document.getElementById('cancel-upload') as HTMLElement
```

Comment:
Might be a good use of the `expect` function that's been put into our `utils` files for the client and the server (may need to use the server version, since the client version contains a vscode api call). Here's the function:
```
function expect<T>(value: T | null | undefined, message: string): T {
  if (value == null) {
    throw new Error(message)
  }
  return value
}
```
The gain out of using this is that we'd end up actually checking to see if we can treat these as `HTMLElement` before we start using them and throw an error in the correct place otherwise.
```suggestion
  const dropArea = expect(document.getElementById('drop-area'), 'html file must contain #drop-area')
  const preview = expect(document.getElementById('preview'), 'html file must contain #preview')
  const uploadButton = expect(document.getElementById('trigger-upload'), 'html file must contain #trigger-upload')
  const cancelButton = expect(document.getElementById('cancel-upload'), 'html file must contain #cancel-upload')
```